File: src/lessons/welcome/methods/picture4/MethodsPicture4Entity.java
package lessons.welcome.methods.picture4;

import java.awt.Color;

import jlm.universe.bugglequest.SimpleBuggle;

/* The suppress warning is sometimes mandatory for student code to compile cleanly */
@SuppressWarnings("unused")
public class MethodsPicture4Entity extends SimpleBuggle {

	/* BEGIN TEMPLATE */
	/* BEGIN SOLUTION */
	void mark() {
		brushDown();
		brushUp();
	}

	void squareA(Color c) {
		setBrushColor(c);
		
		forward();
		mark();

		turnLeft();
		forward();

		turnLeft();
		forward();
		mark();

		turnLeft();
		forward();
		turnLeft();
	}
	
	void squareB(Color c) {
		setBrushColor(c);
		mark();

		forward();

		turnLeft();
		forward();
		mark();

		turnLeft();
		forward();

		turnLeft();
		forward();
		turnLeft();
	}

	void bigSquare() {
		squareA(Color.red); 
		forward(2);
		squareB(Color.blue);
		backward(2);
		turnLeft();
		forward(2);
		turnRight();
		squareB(Color.yellow);
		forward(2);
		squareA(Color.green);
		
		backward(2);
		turnLeft();
		backward(2);
		turnRight();
	}
	
	public void run() {
		bigSquare();
		forward(4);
		bigSquare();
		
		backward(4);
		turnLeft();
		forward(4);
		turnRight();
		
		bigSquare(); 
		forward(4);
		bigSquare();
	}
	/* END SOLUTION */
	/* END TEMPLATE */
}

File: src/lessons/welcome/methods/picture4/MethodsPicture4Entity.py
import java.awt.Color as Color

# BEGIN SOLUTION
def mark():
   brushDown()
   brushUp()

def squareA(c):
   setBrushColor(c)
   forward()
   mark()
   left()
   forward()
   left()
   forward()
   mark()
   left()
   forward()
   left()
   
def squareB(c):
   setBrushColor(c)
   mark()
   forward()
   left()
   forward()
   mark()
   left()
   forward()
   left()
   forward()
   left()

def bigSquare():
   squareA(Color.red)
   forward(2)
   squareB(Color.blue)
   backward(2)
   left()
   forward(2)
   right()
   squareB(Color.yellow)
   forward(2)
   squareA(Color.green)
   backward(2)
   left()
   backward(2)
   right()
	
bigSquare()
forward(4)
bigSquare()
		
backward(4)
left()
forward(4)
right()
		
bigSquare() 
forward(4)
bigSquare()
# END SOLUTION

